### ISSUE TITLE
ObjectMapper serialization for boolean fields named isFieldName changed between versions 2.10.5 and 2.12.4

### ISSUE THREAD

**Issue by kyleboon** [FasterXML/jackson-databind#3345]
**Describe the bug**
I'm upgrading a project from Jackson 2.10.5 to 2.12.4 and one of the classes started serializing differently. The class has a field named `isOpen` which is not a boolean. In 2.10.5 this serialized as `"isOpen": { }` In 2.12.4 the field is dropped entirely. 

**Version information**
2.12.4

**To Reproduce**

This first test passes with Jackson 2.10.5 and fails in 2.12.4. The second test passes in both versions.

```
class JacksonTest {
    @Test
    fun `will serialize a data class with a property named isOpen if it is not a boolean`() {
        data class Thing(val isOpen: Map<String, Boolean>)

        val data = Thing(isOpen = mapOf(Pair("a", true)))

        ObjectMapper().writeValueAsString(data) shouldBe """{"isOpen":{"a":true}}"""
    }

    @Test
    fun `will serialize a data class with a property named isOpen if it is a boolean`() {
        data class Thing(val isOpen: Boolean)

        val data = Thing(isOpen = true)

        ObjectMapper().writeValueAsString(data) shouldBe """{"open":true}"""
    }
}
```

**Comment by cowtowncoder** [comment]
Is this with Kotlin? If so, the right project is `jackson-module-kotlin`, not `jackson-databind` -- I can transfer but want to make sure first. Kotlin's logic for "is propeties" differs from Java Beans spec and there have been changes in this area.

**Comment by sudo-nan0-RaySK** [comment]
Same happening with Java as well.

**Comment by cowtowncoder** [comment]
@sudo-nan0-RaySK I don't think this should be the case, unless Kotlin module is registered. If it does occur, this would belong here, but for now I assume it is Kotlin-MODULE specific (not necessarily language/value however; but module does not discriminate).
